macros/src/parser/state_machine.rs

```diff
@@ -117,6 +119,14 @@ impl parse::Parse for StateMachine {
                     statemachine.events_attr = Attribute::parse_outer(input)?;
                 }
 
+                "entry_exit_async" => {
+                    input.parse::<Token![:]>()?;
+                    let entry_exit_async: syn::LitBool = input.parse()?;
```

Comment:
Don't we handle this by having `async` as a key word in the DSL for guards and actions? I would imagine that would be the same for on_entry and on_exit? I'm trying to avoid adding in these key-value pairs where possible because it was getting a bit unweidly

Reply:
the async is declared in the transitions scope whereas the on_entry and on_exit is default behaviour, I dont see a way how we could do this differently, maybe with async_on_entry_* but not sure how we would detect when process_event would require to be async like so

but I am open to suggestions if you think of something differently.

in cpp version, boost ext sml

this is the lingo

```
, "s1"_s + sml::on_entry<_> / [] { std::cout << "s1 on entry" << std::endl; }
```

it could somehow added to the transitions, to explicitly declare the on_entry/on_exit actions, but I am not sure it is better